File: src/main/scala/cn/netty/demo/ClientHandler.scala
package cn.netty.demo

import io.netty.channel.{ChannelHandlerContext, ChannelInboundHandlerAdapter}

/**
  * Created by erjun on 2017/3/17.
  */
class ClientHandler extends ChannelInboundHandlerAdapter{

	override def exceptionCaught(ctx: ChannelHandlerContext, cause: Throwable): Unit = {
		println("server exceptionCaught")
		ctx.close()
	}

	override def channelActive(ctx: ChannelHandlerContext): Unit = {
		println("client active and write hello world")
		ctx.writeAndFlush("hello world")

	}

	override def channelRead(ctx: ChannelHandlerContext, msg: scala.Any): Unit = {
		msg match {
			case message: String => println(s"string:  $message")
			case NettyMsg(message) => println(s"string: $message")
		}
	}

	override def channelRegistered(ctx: ChannelHandlerContext): Unit = {
		println("client registered")
	}

	override def channelReadComplete(ctx: ChannelHandlerContext): Unit = {
		println("client read complete")
	}

}

File: example/src/main/java/io/netty/example/nettySimple/ClientHandler.java
package io.netty.example.nettySimple;

import io.netty.channel.ChannelHandlerContext;
import io.netty.channel.ChannelInboundHandlerAdapter;
import io.netty.util.AttributeKey;

/**
 * Created by erjun on 2017/9/1.
 */
public class ClientHandler  extends ChannelInboundHandlerAdapter {

	@Override
	public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
		// 下面，将接收到的值，写到通道里
		ctx.channel().attr(AttributeKey.valueOf("ChannelKey")).set(msg.toString());
		// 短连接，获取完成后，关闭连接
		// 如果不关闭的话，客户端是获取不到数据的
		ctx.channel().close();
	}
}
